redaxo/src/addons/mediapool/pages/index.php

```diff
@@ -58,6 +58,24 @@
 
 if (-1 == $rex_file_category) {
     $rex_file_category = rex_session('media[rex_file_category]', 'int');
+
+    // check permission not given get first there the user have
+    if ($rex_file_category >= 0 && !rex::getUser()->getComplexPerm('media')->hasAll()) {
+        if (!rex::getUser()->getComplexPerm('media')->hasCategoryPerm($rex_file_category)
+            && !rex::getUser()->getComplexPerm('media_read')->hasCategoryPerm($rex_file_category)
+        ) {
+            $rex_file_category = 0;
+            $firstId = rex::getUser()->getComplexPerm('media')->getFirstId();
+            $firstReadId = rex::getUser()->getComplexPerm('media_read')->getFirstId();
+
+            if ($firstId > $firstReadId) {
+                $firstId = $firstReadId;
```

Comment:
Mich wundert, dass hier Kategorie-IDs nach der Größe verglichen werden.
Die Mediacats werden ja alphabetisch sortiert, eine höhere ID kann in der Struktur also auch "früher" kommen:

![Screenshot 2020-06-09 22 49 24](https://user-images.githubusercontent.com/330436/84198268-91c1a680-aaa3-11ea-9578-9444a401da62.png)

Reply:
Hier geht es ja nicht um die kategorien aus der DB. Sonder um die erste in der liste der permissions. Es link sofern man beim aufruf des MP durch ein media widget dann in entweder die erste kategorie aus media oder media_read je nachdem welche eben die kleinere ID hat. Man muss das nicht so machen, wurde hier halt so gemacht... Man könnte sich auch an `$rex_file_category` orientieren und entsprechend die nächste mögliche id nehmen die eben am dichtesten an der ursprungsid liegt.

Reply:
Ein User mit eingeschränkten Kategorien, wo landet er initial im MP? Greift da auch diese Logik von hier? Oder sieht der einen leeren Pool und muss dann erst eine Kategorie auswählen?
Falls letzteres: Warum dann nicht gleiches Verhalten bei "ungültiger" Kategorie-ID?
Oder ggf. könnte man auch einen Fehler ausgeben stattdessen.